### ISSUE TITLE
transformers GPT2Model does not take 3d attention_mask

### ISSUE THREAD

**Issue by bzantium** [lassl/lassl#15]
GPT2Model from huggingface transformers assume that attention_mask is only 2D tensor mask.
https://github.com/huggingface/transformers/blob/9490d668d2f59ad2e7a4db3dc7ed2f9684af369c/src/transformers/models/gpt2/modeling_gpt2.py#L697
![image](https://user-images.githubusercontent.com/19511788/123636282-a102c600-d857-11eb-8e06-2034efb63a8b.png)

However, if an instance is made of two different document sources, latter one should not attend the former one when LM pretraining. To solve this, attention_mask of 3D tensor mask should be injected.
